include/quicr/fetch_track_handler.h

```diff
@@ -0,0 +1,164 @@
+        {
+            return std::shared_ptr<FetchTrackHandler>(
+              new FetchTrackHandler(full_track_name, priority, group_order, start_group, end_group));
+        }
+
+        /**
+         * @brief Get the status
+         *
+         * @returns The status of the fetch
+         */
+        constexpr Status GetStatus() const noexcept { return status_; }
+
+        /**
+         * @brief Notification of received [full] data object
+         *
+         * @details Event notification to provide the caller the received full data object
+         *
+         * @warning This data will be invalided after return of this method
+         *
+         * @param object_headers    Object headers, must include group and object Ids
+         * @param data              Object payload data received, **MUST** match ObjectHeaders::payload_length.
+         */
+        virtual void ObjectReceived([[maybe_unused]] const ObjectHeaders& object_headers,
+                                    [[maybe_unused]] BytesSpan data)
+        {
+        }
+
+        /**
+         * @brief Notification of a partial object received data object
+         *
+         * @details Event notification to provide the caller the received data object
+         *
+         * @warning This data will be invalided after return of this method
+         *
+         * @param object_headers    Object headers, must include group and object Ids
+         * @param data              Object payload data received, can be <= ObjectHeaders::payload_length
+         */
+        virtual void PartialObjectReceived([[maybe_unused]] const ObjectHeaders& object_headers,
+                                           [[maybe_unused]] BytesSpan data)
+        {
+        }
+
+        /**
+         * @brief Notification of subscribe status
+         *
+         * @details Notification of the subscribe status
+         *
+         * @param status Indicates status of the subscribe
+         */
+        virtual void StatusChanged([[maybe_unused]] Status status) {}
+
+        /**
+         * @brief Get priority
+         *
+         * @return Priority value
+         */
+        constexpr messages::ObjectPriority GetPriority() const noexcept { return priority_; }
+
+        /**
+         * @brief Get group order
+         *
+         * @return GroupOrder value
+         */
+        constexpr messages::GroupOrder GetGroupOrder() const noexcept { return group_order_; }
+
+        constexpr messages::GroupId GetStartGroup() const noexcept { return start_group_; }
+        constexpr messages::GroupId GetEndGroup() const noexcept { return end_group_; }
+
+      private:
+        /**
+         * @brief Set the status
+         * @param status Status of the subscribe
+         */
+        void SetStatus(Status status) noexcept
+        {
+            status_ = status;
+            StatusChanged(status);
+        }
+
+        Status status_{ Status::kObjectNotAvailable };
```

Comment:
Default should probably be `kPendingFetchResponse` upon construction.  Transport will update it, but at least for the app, it'll know that it's still waiting on the transport to do something.